### DISCUSSION TITLE
Similarity thresholds

### DISCUSSION THREAD

**Discussion by PARODBE** [rdkit/rdkit#6185] [Q&A]
Hi,

I hace a question and It would be great if someone can help me. If you want Split you drugs by similarity spaces (low, intermediate and high) and you are using Morgan fingerprints with 2048 bits and radius = 2, how would you do this? I know that similarity with Morgan is less than rdkit, or that with more bits less similarity etc...but do you know any article or blog about this?

Thanks
Pablo

**Comment by greglandrum** [comment]
@PARODBE I have done multiple RDKit blog posts on the topic of similarity thresholds.
Here are a couple of them:
https://greglandrum.github.io/rdkit-blog/posts/2021-05-21-similarity-search-thresholds.html
https://greglandrum.github.io/rdkit-blog/posts/2021-05-18-fingerprint-thresholds1.html

**Reply by PARODBE** [reply]
Thank you very much @greglandrum for your response and for links, they are very useful. I understand that your work shows the number of compounds you get according to each threshold, and that a stricter threshold will produce fewer similar compounds, while a more relaxed one will produce more. Similarly, decreasing the number of bits will increase similarity (as will testing based on radius or type of fingerprints). However, I remain unsure how to determine the threshold for very similar compounds when using morganfingerprints with a radius of 2 and 2048 bits (for example). I understand that probable there is not a gold rule...

**Reply by greglandrum** [reply]
It really depends on what you mean by "very similar" and what your overall goals are.
If your objective is to only retrieve very similar compounds, then I'd suggest setting the threshold quite high: 0.8 or 0.9. At those levels almost everything you get should look quite similar to the query. I don't have a blog post on this because I haven't figured out a good automated way to test it yet (though maybe I could do something with molecular matched pairs and/or MCS). 
The thresholds blog post (https://greglandrum.github.io/rdkit-blog/posts/2021-05-21-similarity-search-thresholds.html) has a focus on retrieiving as many closely related compounds as possible (with closely related defined in the post).

**Comment by bp-kelley** [comment]
@PARODBE when in doubt, look at the distribution of similarities.

this paper has been my goto for a while:

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2914517/

The warning, however, is that chemists don’t think in fingerprint similarity space so you need to factor in your use case.
